doc/ref_arch/kubernetes/chapters/appendix-a.md

```diff
@@ -1,21 +1,25 @@
 [<< Back](../../kubernetes)
-# Appendix A - Guidance for VNF-to-CNF Transition
+# Appendix A - Guidelines
 <p align="right"><img src="../figures/bogo_ifo.png" alt="scope" title="Scope" width="35%"/></p>
 
 ## Table of Contents
-* [A.1 TBC](#A.1)
-* [A.2 TBC](#A.2)
-* [A.3 TBC](#A.3)
-* [A.4 TBC](#A.3)
+* [A.1 Guidance for Vendors](#A.1)
+* [A.2 RA Transitional Generations](#A.2)
+* [A.3 Guidance for VNF-to-CNF Transition](#A.3)
+* [A.4 TBC](#A.4)
 
 <a name="A.1"></a>
-## A.1 TBC
+## A.1 Guidance for Vendors
 
 <a name="A.2"></a>
-## A.2 TBC
+## A.2 RA Transitional Generations
+
+CNF Generation are defined by [Adoption Governance WS](../../../doc/gov/chapters/chapter08.md#844-cnf-transition-generations)
+
+> This will contains architectural choices that are suitable for each generation based on the Exceptions (deviation) agreed on in RM A.4
 
 <a name="A.3"></a>
-## A.3 TBC
+## A.3 Guidance for VNF-to-CNF Transition
```

Comment:
This is a bit misleading as CNF-s are also VNF-s. I would rather use Guideance for cloud native architecture or something like this. 
Also, I'm not sure if it is part of the scope of CNTT to discuss VNF internal architecture.

Reply:
I don't agree that CNFs are VNFs - we surely need to distinguish between modern cloud native software (CNFs) and the current standard virtualised software (VNFs). I think it is helpful to keep them separate and show a transition from today to the future where both exist.

Reply:
Our definition for CNF:
Cloud native Network Function (CNF): An implementation of a Virtual Network Function (VNF, as defined by ETSI GS NFV 003) that adheres to the CNCF Cloud Native Definition or a VNF that is transitioning to cloud native.
If we drop that, we deviate completely from ETSI and have to write a new NFVO.

Reply:
Ah OK, I hadn't noticed that.  Worth saying that this is not the definition being shaped in the CNCF TUG any more - we've moved away from CNF as being a type of VNF and defining as a distinct type of it's own.
https://docs.google.com/document/d/1LO6G4sWHJKKW77z7gj2Aqt3IF3dRJL5q6Q1wXI4Kj8g/edit?pli=1#heading=h.mk8qm2iyy17n

Reply:
> If we drop that, we deviate completely from ETSI and have to write a new NFVO.

Not sure I see the logic here - why would you need a different NFVO?

Reply:
It will be very difficult using "CNF" if the definition by CNCF TUG is changing. We really derived our definition from there. Are our references to the CNCF TUG whitepaper now outdated? 
I think it is important to say that a CNF is a VNF. NFVO uses templates for the orchestration. A VNF has a VNFD that specifies important characteristics for e.g. scalability and reliability. NS and the related templates are defined to orchestrate VNFs, their networking and connection to PNFs. We should not try to invent that again. It is true that not everything in VNFD easily maps to container technology. ETSI NFV is working on how to do it.

Reply:
> It will be very difficult using "CNF" if the definition by CNCF TUG is changing. We really derived our definition from there. Are our references to the CNCF TUG whitepaper now outdated?

To be clear, it has changed but TUG is looking to fix on a definition and continue to change it.

> We should not try to invent that again. 

I disagree with this and I would hope that NFV v4 from ETSI will reflect this, and if that means the NFVO needs to change to be able to orchestration network services that consist of VNFs and CNFs then that's ok, for me - again, that is something I expect the next iteration of ETSI NFV to define, rather than looking for existing models.  

I strongly feel that the whole point of cloud native is to start again from scratch / build from the ground up / etc. to take advantage of cloud principles and technologies.  This affects everything from the design through to the operating models for development, delivery and operations.  It has to be distinct from a "virtualised" application which relies on existing processes otherwise we will continue to see the "lift-and-shift" approach that has largely take place with NFV - i.e. we will end up with Pods that are requesting 32vCPU, and years between upgrades etc. instead of a true reassessment of the data model of each application and decomposition of functionality based on that to allow for the realisation of cloud native benefits.

Reply:
I agree with @ulikleber that we should not redefine the whole MANO stack for cloud native VNF-s and state, that a CNF is a VNF. It is not about the internal architecture of the applications or the resource usage of the smallest manageable resource unit, but about the high level LCM of the apps.
I believe that both ALT4.6, ALT4.7 and ALT4.8 of the CNCF TUG CNF definitions also reflects this and do not try to define a different way of network level orchestration for CNFs from the way how VNF-s are orchestrated.

Reply:
To be clear, I don't think MANO needs completely redefining. But o you point @CsatariGergely I do feel that the high level LCM of a VNF (non-immutability, imperative config mgmt, months/years between updates, etc.) is, and must be, different to a cloud native NF (100% immutable, declarative config mgmt, days/weeks between updates, etc.). Otherwise, what's the point, as I say we're just lifting and shifting from Physical to VM to container with none of the benefits of cloud native?